src/lib/lore.sh

```diff
@@ -352,29 +462,102 @@ function processing_new_patches()
-        processed_line="$(process_name "$line")${SEPARATOR_CHAR}"
+        processed_patchset="$(process_name "$line")${SEPARATOR_CHAR}"
         ;;
       1) # EMAIL
-        processed_line+="${line}${SEPARATOR_CHAR}"
+        processed_patchset+="${line}${SEPARATOR_CHAR}"
         ;;
       2) # TITLE
-        title="$line"
+        patch_title="$line"
         ;;
     esac
 
     ((count++))
 
-  done <<< "$pre_processed"
+  done <<< "$pre_processed_patches"
   wait
 
-  # From the last interaction, we have one extra index that does not exist
-  ((index--))
-
-  for i in $(seq 0 "$index"); do
+  for i in $(seq "$starting_index" "$((PATCHSETS_PROCESSED - 1))"); do
     list_of_mailinglist_patches["$i"]=$(< "${shared_dir_for_parallelism}/${i}")
   done
 }
 
+# This function is primarily a mediator to manage the complex action of fetching
+# the lastest patchsets from a public mailing list archived on lore.kernel.org.
+# The fetching of patchsets has 3 steps:
+#  1. Build a lore query URL to match only messages that are patches from a given
+#     timeframe.
+#  2. Make a request to the URL built in step to obtain a list of patches from
+#     the respective timeframe ordered by the recieved time of the patch on the
+#     lore.kernel.org servers.
+#  3. Process the list of patches to a list of patchsets stored in the
+#     `list_of_mailinglist_patches` array.
+#
+# In case the number of patchsets in `list_of_mailinglist_patches` is less than
+# `NUMBER_OF_PATCHSETS_PER_FETCH`, get a adjacent timeframe chunk and repeat steps
+# 1 to 3.
+#
+# Each entry in `list_of_mailinglist_patches` has the following patchset metadata
+# separated by `SEPARATOR_CHAR`:
+#   author name, author email, patchset version, number of patches, patch title, message-ID
+#
+# @target_mailing_list: A string name that matches the mailing list name
+#   registered to lore
+# @flag: Flag to control function output
+#
+# Return:
+# If either step 1 or 2 fails, returns the error code from these steps, and 0, otherwise.
+function fetch_latest_patchsets_from()
+{
+  local target_mailing_list="$1"
+  local flag="$2"
+  local raw_xml
+  local lore_query_url
+  local days
+  local last_patch_timestamp
+  local xml_result_file_name
+  local pre_processed_patches
+  local ret
+
+  # Assigning starting values.
+  days="$TIMEFRAME_IN_DAYS"
+  last_patch_timestamp=''
+  xml_result_file_name="${target_mailing_list}-patches.xml"
+  PATCHSETS_PROCESSED=0
+  flag=${flag:-'SILENT'}
+
+  while [[ "$PATCHSETS_PROCESSED" -lt "$NUMBER_OF_PATCHSETS_PER_FETCH" ]]; do
+    # Building URL for querying lore servers for a xml file with patches.
+    lore_query_url=$(compose_lore_query_url_with_verification "$target_mailing_list" "$days" "$last_patch_timestamp")
+    ret="$?"
+    [[ "$ret" != 0 ]] && return "$ret"
+
+    # Request xml file with patches.
+    download "$lore_query_url" "$xml_result_file_name" "$CACHE_LORE_DIR" "$flag"
+    ret="$?"
+    [[ "$ret" != 0 ]] && return "$ret"
```

Comment:
In the same line of my last response, a bad URL means bad URLs for any of the next iterations of the while loop. However, I agree that if a download failed because of network issues, for example, we should retry to download it. Do you have an idea to fine grain those errors to treat them accordingly?